src/web/mjs/connectors/LeerCapitulo.mjs

```diff
@@ -0,0 +1,54 @@
+import AnyACG from './templates/AnyACG.mjs';
+export default class LeerCapitulo extends AnyACG {
+    constructor() {
+        super();
+        super.id = 'leercapitulo';
+        super.label = 'LeerCapitulo';
+        this.tags = [ 'manga', 'spanish' ];
+        this.url = 'https://www.leercapitulo.com';
+        this.categories = ['/publicandose/', '/finalizado/', '/pausado/', '/cancelado/'];
+        this.path ='/status';
+        this.language = 'es';
+        this.queryChapters = 'div.chapter-list ul li h4 a';
+        this.queryPages = 'div.chapter-content-inner p#arraydata';
+    }
+    async _getMangas() {
+        let mangaList = [];
+        for (const cat of this.categories) {
+            const uri = new URL(this.path + cat, this.url);
+            const request = new Request(uri, this.requestOptions);
+            const data = await this.fetchDOM(request, 'ul.pagination li:last-of-type a');
+            const pageCount = data.length > 0 ? parseInt(data[0].href.match(/(\d)+$/)[1]) : 1;
+            for(let page = 1; page <= pageCount; page++) {
+                const mangas = await this._getMangasFromPage(page, cat);
+                mangaList.push(...mangas);
+            }
+        }
+        return mangaList;
+    }
+    async _getMangasFromPage(page, cat) {
+        const uri = new URL(this.path + cat+ '?page='+ page, this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'div.cate-manga div.media-body > a');
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.title.trim()
+            };
+        });
+    }
+    async _getPages(chapter) {
+        let counter = 0;
+        let chapters = [];
+        while (chapters.length == 0 && counter < 3) {
+            try {
+                await this.wait(500);
+                chapters = await super._getPages(chapter);
```

Comment:
_getPages being a promise you can .catch instead of try (easier to read)

Reply:
How would you do that regarding the previous requested change? 

```
    async _getPages(chapter) {
        let counter = 0;
        let chapters = [];
        for (let i = 1; i <= 3; i++) {
            await this.wait(500);
            chapters = await super._getPages(chapter);
            if (counter.length > ) break;
        }
        return chapters;
    }
```

Reply:
something like
```javascript
        for (let retries = 0; retries < 3; retries ++) {
            await this.wait(500);
            chapters = await super._getPages(chapter)
            .catch(() => { /* ignore */ })
            .then( pages => return pages );
            if (chapters .length > 0) return chapters;
        }}
        return chapters;
```
but now that i see that one, try catch is probably better